### PR TITLE
Us profile page

### PR DESCRIPTION
**Pull request by MareikeFla**
you can log in using your gitHub account, reach a profile page using the menu and edit your profile.

### CODE REVIEW

File: components/Profile/ProfileForm.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      >
+        {editableInfo.map((info) => {
+          const { key, text, value, isRequired } = info;
+          return (
+            <FormSection key={key}>
+              <FormLabel htmlFor={key}>
+                {text} {isRequired && "*"}
+              </FormLabel>
+              <FormInput
+                required={key === "name"}
```
Comment (JessicaLoers) [inline components/Profile/ProfileForm.js:48]: I wonder why you don't use the isRequired flag directly here. But perhaps there is a reason for this. 🙃
```suggestion
                required={isRequired}
```

File: lib/profile/profileHelper.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Array with user infos is used for mapping to reduce repitation
+const pictureSize = 128;
+
+export function formatedUserInfo(session) {
+  const profilePicture = {
+    key: "image",
+    text: "Profilbiild",
```
Comment (JessicaLoers) [inline lib/profile/profileHelper.js:7]: Just a typo:
```suggestion
    text: "Profilbild",
```

File: lib/useData.js [hunk 3]
```diff
@@ -151,7 +144,24 @@ export const DataProvider = ({ children }) => {
...
+      },
+      body: JSON.stringify(userData),
+    });
+
+    if (response.ok) {
+      return true;
+    } else if (!response.ok) {
+      return false;
+    }
+  }
```
Comment (JessicaLoers) [inline lib/useData.js:164]: - Ensure the default form submission behaviour is prevented

```suggestion
  async function updateUser(event, user) {
   event.preventDefault();
    const data = new FormData(event.target);
    const formData = Object.fromEntries(data);
    const userData = { ...user, ...formData };
    const response = await fetch(`/api/user/${userData.id}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(userData),
    });

    if (response.ok) {
      return true;
    } else if (!response.ok) {
      return false;
    }
  }
```

File: pages/profile/index.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import ArrowButton from "@/components/ArrowButton/ArrowButton";
+import Profile from "@/components/Profile/Profile";
+import ProfileForm from "@/components/Profile/ProfileForm";
+import { signIn, useSession } from "next-auth/react";
+import Loading from "@/components/Loading/Loading";
+import { useData } from "@/lib/useData";
+import { useState } from "react";
+import { formatedUserInfo } from "@/lib/profile/profileHelper";
+
+export default function profilePage() {
```
Comment (JessicaLoers) [inline pages/profile/index.js:11]: Please use PascalCase for component names.
```suggestion
export default function ProfilePage() {
```

File: pages/profile/index.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const wasSuccsessfull = await updateUser(event, session.user);
+    if (wasSuccsessfull) {
+      updateSession();
+    }
+    toggleEditMode();
+    return wasSuccsessfull;
+  };
```
Comment (JessicaLoers) [inline pages/profile/index.js:28]: Typo:
```suggestion
    const wasSuccsessful = await updateUser(event, session.user);
    if (wasSuccsessful) {
      updateSession();
    }
    toggleEditMode();
    return wasSuccsessful;
  };
```

File: pages/profile/index.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  if (!session) {
+    return (
+      <MessageCard>
+        <p>Du bist nicht berechtigt diese Seite zu besuchen.</p>
+        <ArrowButton onClick={() => signIn()}>Jetzt Anmelden</ArrowButton>
+      </MessageCard>
+    );
+  } else if (!editMode) {
+    return (
+      <Profile toggleEditMode={toggleEditMode} userInfo={userInfo}></Profile>
```
Comment (JessicaLoers) [inline pages/profile/index.js:44]: Could be self-closing. 
```suggestion
      <Profile toggleEditMode={toggleEditMode} userInfo={userInfo} />
```

File: pages/profile/index.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    );
+  } else if (editMode) {
+    return (
+      <ProfileForm
+        toggleEditMode={toggleEditMode}
+        userInfo={userInfo}
+        handleSubmit={handleSubmit}
+      ></ProfileForm>
```
Comment (JessicaLoers) [inline pages/profile/index.js:52]: Could be self-closing.
```suggestion
      <ProfileForm
        toggleEditMode={toggleEditMode}
        userInfo={userInfo}
        handleSubmit={handleSubmit}
      />
```

**Review by JessicaLoers — APPROVED** [review]
**hey team ✨**
have very few comments, which you're welcome to look at. 
**Really well done.**

File: components/Profile/ProfileForm.js (hunk 1 above)
Comment (MareikeFla) [reply components/Profile/ProfileForm.js:48]: Yesterday evening, after posting the PR I saw that and also wondered why I did not do it 🙃

**Pull request merged**